mirascope/core/base/_create.py

```diff
@@ -196,10 +201,15 @@ def inner(
                 *args: _P.args, **kwargs: _P.kwargs
             ) -> TCallResponse | _ParsedOutputT:
                 fn_args = get_fn_args(fn, args, kwargs)
-                dynamic_config = get_dynamic_configuration(fn, args, kwargs)
+                if (
+                    dynamic_config := get_dynamic_configuration(fn, args, kwargs)
+                ) and "client" in dynamic_config:
```

Comment:
pyright requests "client" in dynamic_config for TypedDict

Reply:
why do we check this here but check for dict instance in async? why not the same?

Reply:
The code works fine.
```python
dynamic_config = await get_dynamic_configuration(fn, args, kwargs)
                if dynamic_config is not None:
                    dynamic_client = dynamic_config.get("client")
                else:
                    dynamic_client = None
```

But, the code got a type error.
```python
                dynamic_config = await get_dynamic_configuration(fn, args, kwargs)
                if dynamic_config:
                    dynamic_client = dynamic_config.get("client")
                else:
                    dynamic_client = None
```

```
/Users/koudai/PycharmProjects/mirascope/mirascope/core/base/_create.py
  /Users/koudai/PycharmProjects/mirascope/mirascope/core/base/_create.py:166:53 - error: "get" is not a known attribute of "None" (reportOptionalMemberAccess)
1 error, 0 warnings, 0 informations 

```

I think `if dynamic_config` is `True` guarantee `dynamic_config` is not None. But, the result is not same :(
pyright said this is `None`